### ISSUE TITLE
CatalogCategoryTree: String cannot represent non scalar value: array

### ISSUE THREAD

**Issue by alexandersch** [vrann/m2graphql#1]
Hi Vrann,

When executing a query on MagentoCatalogCategoryTree, I get the following error:

![schermafbeelding 2017-09-27 om 11 02 15](https://user-images.githubusercontent.com/3180083/30904946-bb57a164-a373-11e7-9212-5fdb6ff22544.png)

Are you planning on fixing this in the near future?

**Comment by vrann** [comment]
@Alexandersch good catch, this is caused by the fact that GraphQL does not allow to have recursive objects, while Category Tree is recursive. A temporal fix was to substitute the reference to the children category with the object of type String. A better fix would be to allow just limited level of recursion.
